src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3104,11 +3108,215 @@ public override IOperation VisitObjectCreation(IObjectCreationOperation operatio
             ImmutableArray<IArgumentOperation> visitedArgs = VisitArguments(operation.Arguments);
 
             // Initializer is removed from the tree and turned into a series of statements that assign to the created instance
-            var objectCreation = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+            IOperation initializedInstance = new ObjectCreationExpression(operation.Constructor, initializer: null, visitedArgs, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+
+            if (operation.Initializer != null)
+            {
+                SpillEvalStack();
 
-            // PROTOTYPE(dataflow): For a non-null initializer, we'll need to assign the created object to a flow reference and rewrite all initializers to assign to/call functions on it, then return return the flow reference
+                int initializerCaptureId = _availableCaptureId++;
+                AddStatement(new FlowCapture(initializerCaptureId, initializedInstance.Syntax, initializedInstance));
 
-            return objectCreation;
+                initializedInstance = new FlowCaptureReference(initializerCaptureId, initializedInstance.Syntax, initializedInstance.Type, initializedInstance.ConstantValue);
+                HandleObjectOrCollectionInitializer(operation.Initializer, initializedInstance);
+            }
+
+            return initializedInstance;
+        }
+
+        private void HandleObjectOrCollectionInitializer(IObjectOrCollectionInitializerOperation initializer, IOperation initializedInstance)
+        {
+            // PROTOTYPE(dataflow): Handle collection initializers
+            IOperation previousInitializedInstance = _currentInitializedInstance;
+            _currentInitializedInstance = initializedInstance;
+
+            foreach (IOperation innerInitializer in initializer.Initializers)
+            {
+                handleInitializer(innerInitializer);
+            }
+
+            _currentInitializedInstance = previousInitializedInstance;
```

Comment:
>_currentInitializedInstance = previousInitializedInstance; [](start = 12, length = 58)

If this is the end of the code in this function, consider adding an explicit return statement after it, this would help readability.